scripts/test/create-cluster.sh

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+TEST_DIR=$(cd "$(dirname "$0")" || exit 1; pwd)
+TEMPLATE_FILE="$TEST_DIR/template/eksctl/eks-cluster.yaml"
+
+source "$PWD/scripts/lib/common.sh"
+
+USAGE=$(cat << 'EOM'
+Usage: create-cluster.sh -n [cluster-name]
+Creates an EKS Cluster with Nodegroups. The Cluster is
+created using eksctl with a pre-defined eksctl template.
+ Required:
+ -n   Name of the EKS cluster
+ Optional:
+ -v   K8s Version. Defaults to 1.20
+ -r   Region of the EKS Cluster. Defaults to us-west-2
+ -c   eksctl Cluster config path. Defaults to build/eks-cluster.yaml
+ -t   Type of cluster. Defaults to eksctl cluster. Can be set to "kops" or "eksctl"
+EOM
+)
+
+
+
+while getopts "n:v:r:c:t:" o; do
+  case "${o}" in
+    n) # Name of the EKS Cluster
+      CLUSTER_NAME=${OPTARG}
+      ;;
+    v) # K8s Version of the EKS Cluster
+      K8S_VERSION=${OPTARG}
+      ;;
+    r) # Region where EKS Cluster will be created
+      AWS_REGION=${OPTARG}
+      ;;
+    c) # eksctl cluster config file path
+      CLUSTER_CONFIG_PATH=${OPTARG}
+      ;;
+    t) # Type of cluster to be created
+      CLUSTER_TYPE=${OPTARG}
+      ;;
+    *)
+      echoerr "${USAGE}"
+      exit 1
+      ;;
+  esac
+done
+
+
+
+shift $((OPTIND-1))
+
+if [[ -z "$CLUSTER_NAME" ]]; then
+ echoerr "${USAGE}\n\nmissing: -n is a required flag\n"
+ exit 1
+fi
+
+if [[ -z "$K8S_VERSION" ]]; then
+  K8S_VERSION="1.20"
```

Comment:
As and when a given k8s version goes out of upstream support window, we need to come back and fix this line of code. Would be good if we have better automation to get upstream supported version rather than hard coding the value here

Reply:
I don't see any EKS CLI to get currently supported k8s version. https://docs.aws.amazon.com/cli/latest/reference/eks/index.html

Reply:
I think I can pull latest release version from the GitHub repo https://github.com/kubernetes/kubernetes. But I'm not sure if we use the latest version for our clusters in the current scripts. This is what I observed, although I may be wrong.